frontend-new/components/LinkCreationForm.tsx

```diff
@@ -0,0 +1,172 @@
+import styled from '@emotion/styled'
+import EastRoundedIcon from '@mui/icons-material/EastRounded'
+import { Button, TextField, Typography } from '@mui/material'
+import { useRouter } from 'next/router'
+import { ChangeEvent, FormEvent, useCallback, useMemo, useState } from 'react'
+import { useEffect, useRef, FC } from 'react'
+
+import { ScreenType, useSaveLink, useWindowWidth } from 'app/hooks'
+import { media } from 'app/styles/theme'
+import { LinkCreate, LinkCreateResponse } from 'app/types'
+
+const StyledForm = styled.form`
+  display: grid;
+  grid-template-columns: 5fr 8fr;
+  gap: 8px;
+  grid-template-areas:
+    'a b'
+    'c c';
+`
+
+const Group = styled.div`
+  display: flex;
+  flex-direction: row;
+  background-color: #f4f3ff;
+  border-radius: 32px;
+`
+
+const Cicle = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 32px;
+  height: 32px;
+  background-color: #bdbcf3;
+  border-radius: 32px 0px 0px 32px;
+  color: #fff;
+
+  ${media.TABLET} {
+    width: 48px;
+    height: 48px;
+  }
+
+  ${media.DESKTOP} {
+    width: 64px;
+    height: 64px;
+  }
+`
+
+interface Props {
+  onCreate: ({
+    link,
+    createdResponse,
+  }: {
+    link: LinkCreate
+    createdResponse: LinkCreateResponse
+  }) => void
+}
+
+export const LinkCreationForm: FC<Props> = ({ onCreate }) => {
+  const {
+    query: { sp },
+  } = useRouter()
+  const [formState, setFormState] = useState<LinkCreate>({
+    shortpath: '',
+    destination: '',
+    namespace: 'go',
+  })
+  const saveLink = useSaveLink()
+
+  const shortInpuRef = useRef<HTMLInputElement>(null)
+  const destInpuRef = useRef<HTMLInputElement>(null)
+  const { isMobile } = useWindowWidth()
```

Comment:
Any reason to use this vs. `useMediaQuery`?

Reply:
No reasons, changed, thanks